rapids_build_backend/config.py

```diff
@@ -15,6 +15,7 @@ class Config:
         "allow-nightly-deps": (True, True),
         "build-backend": (None, False),
         "commit-file": ("", False),
+        "dependencies-file": ("dependencies.yaml", True),
```

Comment:
This is hardcoded in dfg, I think -- I don't think we need to make this configurable?

Reply:
I see. I guess the wheel build does not know where the root of the repository is.

Reply:
Actually, I think this discussion uncovers a slightly larger issue. Using rbb with dfg in this way requires dependencies.yaml to be available. While that is always currently true, that need not always be the case. In particular, this will actually break building wheels from sdists because we will not have dependencies.yaml in the wheel. I think the solution we're going to need is to basically do what we do for VERSION files and symlink dependencies.yaml into the root of every Python package directory so that it'll get included in the sdist (there are alternative, more cumbersome solutions if we don't like the symlinks). We should make sure that whatever choice we make doesn't violate dfg's own assumptions about where dependencies.yaml lives.

This is mostly a forward-looking discussion since we don't build sdists right now, but it's worth fixing since otherwise the backend by default breaks some use cases that would otherwise be supported by the upstream backends, and moreover it would prevent us from building with `build`, or really any other valid PEP 517 frontend other than `pip wheel`.

Reply:
Good observation. Yes, symlinking and using a `dependencies.yaml` that is co-located in the same directory as `pyproject.toml` seems like the proper solution.

Reply:
An alternative plan would be for RBB to know its own `pyproject_dir` (in the config?) and subtract that relative path from the paths specified in `dependencies.yaml`. I don't love this idea but thought it was worth mentioning.

Reply:
On further consideration, I think we are in a better position than we thought. sdists would be produced by building the package with the backend, at which point all the textual replacements would already have occurred, i.e. at that point we can safely assume that the dependency list in pyproject.toml is correct. All we have to do is detect that dependencies.yaml is not present and if so not try to update. Since finding the dependencies.yaml file is more dfg's responsibility, we could handle that with a try-except here.

Of course, the idea of sdists opens us up to a much more concerning set of problems where the user takes a libcuspatial-cu12 and builds a wheel from it on a CUDA 11 system and then their dependencies are completely out of whack. That sort of case is something we might want RBB to protect against by adding some logic to the CMake to require a particular version of CUDA when building the sdist. That's something we can do much later though since sdists aren't practically anywhere near the roadmap right now.

Reply:
This is now done.